wallet/wallet.c

```diff
@@ -326,14 +327,25 @@ bool wallet_shachain_load(struct wallet *wallet, u64 id,
 static bool wallet_peer_load(struct wallet *w, const u64 id, struct peer *peer)
 {
 	bool ok = true;
-	sqlite3_stmt *stmt = db_query(__func__, w->db, "SELECT id, node_id FROM peers WHERE id=%"PRIu64";", id);
+	const unsigned char *addrstr;
+
+	sqlite3_stmt *stmt =
+		db_query(__func__, w->db,
+			 "SELECT id, node_id, address FROM peers WHERE id=%"PRIu64";", id);
+
 	if (!stmt || sqlite3_step(stmt) != SQLITE_ROW) {
 		sqlite3_finalize(stmt);
 		return false;
 	}
 	peer->dbid = sqlite3_column_int64(stmt, 0);
 	ok &= sqlite3_column_pubkey(stmt, 1, &peer->id);
+	addrstr = sqlite3_column_text(stmt, 2);
+
+	if (addrstr)
+		parse_wireaddr((const char*)addrstr, &peer->addr, 0);
```

Comment:
Since fmt_wireaddr always formats a port, the 0 here doesn't matter, but it might be good to change it to DEFAULT_PORT in case that changes. I was thinking about removing the port argument but I figured it might be good to keep it in case the address string doesn't have a port which was another use case of this function.

Reply:
> ok &= parse_wireaddr... perhaps?
 
I was weary about that for backward compatibility reasons when coming from an older version of the db with no addresses. Up to you...